provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,183 @@
+ * http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-location.html for implementation
+ * details.
+ */
+@SuppressWarnings("WeakerAccess")
+@RestController
+@RequestMapping(
+  value = {"/api/Location"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+public class LocationController {
+
+  private Transformer transformer;
+
+  private PpmsClient client;
+
+  private Bundler bundler;
+
+  /** Controller setup. */
+  public LocationController(
+      @Autowired LocationController.Transformer transformer,
+      @Autowired Bundler bundler,
+      @Autowired PpmsClient client) {
+    this.transformer = transformer;
+    this.bundler = bundler;
+    this.client = client;
+  }
+
+  private Location.Bundle bundle(MultiValueMap<String, String> parameters, int page, int count) {
+    LocationWrapper locationWrapper = LocationWrapper.builder().build();
+    CareSitesResponse careSitesResponse;
+    List<LocationWrapper> paged = new ArrayList<>();
+    int recordCount = 1;
+
+    if (parameters.get("identifier") != null) {
+      String identifier = parameters.get("identifier").toArray()[0].toString();
+      locationWrapper = client.careSitesById(identifier);
+    } else if (parameters.get("name") != null) {
+      String name = parameters.get("name").toArray()[0].toString();
+      locationWrapper = client.careSitesByName(name);
+    } else {
+      if (parameters.get("city") != null) {
+        String city = parameters.get("city").toArray()[0].toString();
+        careSitesResponse = client.careSitesByCity(city);
+      } else if (parameters.get("state") != null) {
+        String state = parameters.get("state").toArray()[0].toString();
+        careSitesResponse = client.careSitesByState(state);
+      } else {
+        String zip = parameters.get("zip").toArray()[0].toString();
+        careSitesResponse = client.careSitesByZip(zip);
+      }
+      recordCount = careSitesResponse.value().size();
+      int fromIndex = Math.min((page - 1) * count, careSitesResponse.value().size());
+      int toIndex = Math.min((fromIndex + count), careSitesResponse.value().size());
+      paged =
+          IntStream.range(fromIndex, toIndex)
+              .parallel()
+              .mapToObj(i -> client.careSitesByName(careSitesResponse.value().get(i).name()))
+              .collect(Collectors.toList());
+    }
+
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("Location")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(recordCount)
+            .build();
+    return bundler.bundle(
+        BundleContext.of(
+            linkConfig,
+            paged.isEmpty() ? Collections.singletonList(locationWrapper) : paged,
+            transformer,
+            Location.Entry::new,
+            Location.Bundle::new));
+  }
+
+  /** Search by family & given name. */
+  @GetMapping(params = {"city"})
```

Comment:
Shouldn't this be `address-city`? (And `address-state` and `ddress-postalcode` for the other two?)

Reply:
Good catch, I meant to go back and update that to reflect the spec.